update.go

```diff
@@ -265,19 +248,31 @@ other options are ignored.
 					} else {
 						v = -1
 					}
-					*pair.dest = v
+					*pair.dest = &v
 				}
 			}
 
 			r.Pids.Limit = int64(context.Int("pids-limit"))
 		}
 
-		if *r.Memory.Kernel != 0 || *r.Memory.KernelTCP != 0 {
+		// Fix up values
+		if r.Memory.Limit != nil && *r.Memory.Limit == -1 && r.Memory.Swap == nil {
+			// To avoid error "unable to set swap limit without memory limit"
+			r.Memory.Swap = i64Ptr(0)
+		}
+		if r.CPU.Idle != nil && r.CPU.Shares == nil {
+			// To avoid error "failed to write \"4\": write /sys/fs/cgroup/runc-cgroups-integration-test/test-cgroup-7341/cpu.weight: invalid argument"
+			r.CPU.Shares = u64Ptr(0)
+		}
+
+		if (r.Memory.Kernel != nil && *r.Memory.Kernel != 0) || (r.Memory.KernelTCP != nil && *r.Memory.KernelTCP != 0) {
 			logrus.Warn("Kernel memory settings are ignored and will be removed")
```

Comment:
I think it's enough to check for `!= nil` here (as we do ignore `--kernel-memory 0`, but were unable to distinguish it from the default value of 0 before -- now we are able to).

Reply:
Fixed